workbench-core/environments/src/interfaces/listEnvironmentTypeConfigsRequest.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import { z } from 'zod';
+
+// eslint-disable-next-line @rushstack/typedef-var
+export const ListEnvironmentTypeConfigsRequestParser = z
+  .object({
+    envTypeId: z.string(),
+    pageSize: z
+      .string()
```

Comment:
Should we do `. number `?

Reply:
I am following the convention used by other implementations, I think we do this to have more control over the validation message, any thoughts @nguyen102  @aws-tyler

Reply:
The type should match whatever the type should be in the request. If it is coming in as a string (like how it does via query params) thats where the transformation comes in.